Docs/pluggable components

mcandeia opened the pull request:
Thank you for helping make the Dapr documentation better!

> As this feature should work only in UNIX-like systems, should we have commands for windows?

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Please use `dapr init --runtime-version 1.9.0-rc.3` to see it working on standalone or kubernetes mode.

These docs are created under a new "Extending Dapr" section , as `Pluggable Components` is a term that we have been using for a while, I've also added the suffix `gRPC-based` to differentiate. Suggestions are welcome.
 
## Issue reference

<!--Please reference the issue this PR will close: #[issue number]-->

#2799

mcandeia committed:
Add developing Components section

Signed-off-by: Marcos Candeia <marrcooos@gmail.com>

Add .NET initial doc

Signed-off-by: Marcos Candeia <marrcooos@gmail.com>

Remove pluggable component CRD docs

Signed-off-by: Marcos Candeia <marrcooos@gmail.com>

Add zip file with quickstart code

Signed-off-by: Marcos Candeia <marrcooos@gmail.com>

Add using grpc components on standalone mode section

Signed-off-by: Marcos Candeia <marrcooos@gmail.com>

Add dapr runtime version note

Signed-off-by: Marcos Candeia <marrcooos@gmail.com>

Be more impersonal

Signed-off-by: Marcos Candeia <marrcooos@gmail.com>

Draft kubernetes version

Signed-off-by: Marcos Candeia <marrcooos@gmail.com>

Apply tiago's review fixes

Signed-off-by: Marcos Candeia <marrcooos@gmail.com>

Initiual suggestions and begining of the document (#1)

Signed-off-by: Tiago Alves Macambira <tmacam@burocrata.org>

Signed-off-by: Tiago Alves Macambira <tmacam@burocrata.org>

Apply few fixes

Signed-off-by: Marcos Candeia <marrcooos@gmail.com>

pravinpushkar committed:
Updated azure app config metadata fields name (#2830)

* Updated azure app config metadata fields name

Signed-off-by: Pravin Pushkar <ppushkar@microsoft.com>

* Updating index for configuration store

Signed-off-by: Pravin Pushkar <ppushkar@microsoft.com>

Signed-off-by: Pravin Pushkar <ppushkar@microsoft.com>

akhilac1 committed:
postgres configuration store (#2800)

* postgres configuration store

Signed-off-by: akhilac1 <chetlapalle.akhila@gmail.com>

* Added example for subscribe request

Signed-off-by: akhilac1 <chetlapalle.akhila@gmail.com>

* added pool_max_conns to example connection string

Signed-off-by: akhilac1 <chetlapalle.akhila@gmail.com>

* added recommendation to use single subscription with multiple keys

Signed-off-by: akhilac1 <chetlapalle.akhila@gmail.com>

* fixed review comments

Signed-off-by: akhilac1 <chetlapalle.akhila@gmail.com>

Signed-off-by: akhilac1 <chetlapalle.akhila@gmail.com>
Signed-off-by: Mark Fussell <markfussell@gmail.com>
Co-authored-by: Mark Fussell <markfussell@gmail.com>

hhunter-ms committed:
Upmerge v1.8 --> v1.9 - 9/28 (#2839)

* feature/helmchart: supported private helm chart repo to install dapr cluster

Signed-off-by: 1046102779 <seachen@tencent.com>

* Updates wasm middleware documentation with compilation instructions

These changes are coupled with https://github.com/dapr/components-contrib/pull/1833

Signed-off-by: Adrian Cole <adrian@tetrate.io>

* Update daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md

Co-authored-by: Mark Fussell <markfussell@gmail.com>
Signed-off-by: Crypt Keeper <64215+codefromthecrypt@users.noreply.github.com>

* feedback

Signed-off-by: Adrian Cole <adrian@tetrate.io>

* permute

Signed-off-by: Adrian Cole <adrian@tetrate.io>

* feature/helmchart: supported private helm chart repo to install dapr cluster

Signed-off-by: 1046102779 <seachen@tencent.com>

* add note to aws sns/sqs component doc; quick edit

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* change data to value

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* feature/helmchart: supported private helm chart repo to install dapr cluster

Signed-off-by: 1046102779 <seachen@tencent.com>

* Patch 3 (#2691)

* Update release version from 1.8.2 to 1.8.3

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* Update shortcode from 1.8.2 to 1.8.3

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* fix: typo

Signed-off-by: 1046102779 <seachen@tencent.com>

* feature: add a private helm repo for dapr

Signed-off-by: 1046102779 <seachen@tencent.com>

* Update install-dapr-cli.md

Signed-off-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>

* Update daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md

Co-authored-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>
Signed-off-by: yellow chicks <seachen@tencent.com>

* Fix pubsub quickstart component name to match QS source code

Signed-off-by: Nick Greenfield <nigreenf@microsoft.com>

* add output of success

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* remove word

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* add sentence

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* redis 6 update (#2718)

Signed-off-by: yaron2 <schneider.yaron@live.com>

Signed-off-by: yaron2 <schneider.yaron@live.com>

* Update Dapr version to 1.8.4 (#2714)

* Update Dapr version to 1.8.4

Signed-off-by: Nick Greenfield <nigreenf@microsoft.com>

* Update CLI version to 1.8.1

Signed-off-by: Nick Greenfield <nigreenf@microsoft.com>

Signed-off-by: Nick Greenfield <nigreenf@microsoft.com>
Co-authored-by: Yaron Schneider <schneider.yaron@live.com>

* Fixed an issue with code highlighting

Signed-off-by: Lorenzo Montanari <l0ll098@users.noreply.github.com>

* Fix links to broken middleware components

Signed-off-by: Nick Greenfield <nigreenf@microsoft.com>

* RethinkDB has no transactions, etags

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* fix a typo (#2748)

Signed-off-by: Loong <loong.dai@intel.com>

Signed-off-by: Loong <loong.dai@intel.com>

* fix typo (#2751)

Signed-off-by: RcXu <honeyxhl@163.com>

Signed-off-by: RcXu <honeyxhl@163.com>
Co-authored-by: Mark Fussell <markfussell@gmail.com>

* Update component status in v1.7

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Mark components as stable in v1.8

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Correct state azure stablestorage status in v1.7

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Update setup-sqlserver.md

Signed-off-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>

* fix some typos (#2765)

Signed-off-by: Loong <loong.dai@intel.com>

Signed-off-by: Loong <loong.dai@intel.com>

* Typo fix installtion > installation (#2726)

Signed-off-by: Greg. A. <gautric@users.noreply.github.com>

Signed-off-by: Greg. A. <gautric@users.noreply.github.com>
Co-authored-by: Mark Fussell <markfussell@gmail.com>

* Add a new version of the Dapr Diagrams deck (#2773)

Signed-off-by: Nick Greenfield <nigreenf@microsoft.com>

Signed-off-by: Nick Greenfield <nigreenf@microsoft.com>

* Fix typos in MongoDB state configuration file example (#2775)

Closes https://github.com/dapr/docs/issues/2774

Signed-off-by: Yash Nisar <yashnisar@microsoft.com>

Signed-off-by: Yash Nisar <yashnisar@microsoft.com>

* update service invo row item

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* update az cli commands

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* remove unnecessary flags

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* ignore links

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* Fix broken link

Signed-off-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>

* Fix broken link

Signed-off-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>

* remove ignore links

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* java submodule

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* Fix two typo's (#2812)

Signed-off-by: Maarten Mulders <mthmulders@users.noreply.github.com>

Signed-off-by: Maarten Mulders <mthmulders@users.noreply.github.com>

* Updates/fixes to mySQL binding doc (#2819)

* update link (#2823)

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* fix the example code (#2811)

Signed-off-by: Loong <loong.dai@intel.com>

Signed-off-by: Loong <loong.dai@intel.com>

* fix incorrect reference (#2802)

Signed-off-by: Sarthak Sharma <sartsharma@microsoft.com>

Signed-off-by: Sarthak Sharma <sartsharma@microsoft.com>
Co-authored-by: Mark Fussell <markfussell@gmail.com>

* Fix Configuration API unsubscribe endpoint  (#2797)

* fix unsubscribe path

Signed-off-by: Sarthak Sharma <sartsharma@microsoft.com>

* maintain punctuation uniformity

Signed-off-by: Sarthak Sharma <sartsharma@microsoft.com>

* maintain punctuation uniformity

Signed-off-by: Sarthak Sharma <sartsharma@microsoft.com>

* Update configuration_api.md

Making examples consistent

Signed-off-by: Mark Fussell <markfussell@gmail.com>

Signed-off-by: Sarthak Sharma <sartsharma@microsoft.com>
Signed-off-by: Mark Fussell <markfussell@gmail.com>
Co-authored-by: Mark Fussell <markfussell@gmail.com>

* fix link

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* Update go commands to match quickstarts (#2725)

* update go commands

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* ignore links

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>
Co-authored-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>

* Add note for sidecar readiness (#2817)

* Revert "java submodule"

This reverts commit b438550fbb62b4f429f4728a45147209fe8f5d0a.

* revert submodule changes

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* remove note

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* update link

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>
Co-authored-by: Mark Fussell <markfussell@gmail.com>

* Docs contributing guide updates (#2768)

* initial edits

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* updates

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* add templates

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* fix links

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* remove path

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* add docs

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* remove from .github doc

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* fix validate links

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* updates per mark

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* add submodule section

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* remove link and reword

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* fix link?

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* apparently case matters for links now

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* update per mark review

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* update link

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>
Co-authored-by: Mark Fussell <markfussell@gmail.com>

* Changed the broken middleware links. (#2781)

* Add a new version of the Dapr Diagrams deck (#2773)

Signed-off-by: Nick Greenfield <nigreenf@microsoft.com>

Signed-off-by: Nick Greenfield <nigreenf@microsoft.com>
Signed-off-by: Shivani Singh <shivasingh@microsoft.com>

* Fix typos in MongoDB state configuration file example (#2775)

Closes https://github.com/dapr/docs/issues/2774

Signed-off-by: Yash Nisar <yashnisar@microsoft.com>

Signed-off-by: Yash Nisar <yashnisar@microsoft.com>
Signed-off-by: Shivani Singh <shivasingh@microsoft.com>

* Changed the broken middleware links.

Signed-off-by: Shivani Singh <shivasingh@microsoft.com>

* Made the required changes for Rego/OPA

Signed-off-by: Shivani Singh <shivasingh@microsoft.com>

* Fix broken link

Signed-off-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>
Signed-off-by: Shivani Singh <shivasingh@microsoft.com>

* update az cli commands

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>
Signed-off-by: Shivani Singh <shivasingh@microsoft.com>

* remove unnecessary flags

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>
Signed-off-by: Shivani Singh <shivasingh@microsoft.com>

* ignore links

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>
Signed-off-by: Shivani Singh <shivasingh@microsoft.com>

* remove ignore links

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>
Signed-off-by: Shivani Singh <shivasingh@microsoft.com>

* Fix broken link

Signed-off-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>
Signed-off-by: Shivani Singh <shivasingh@microsoft.com>

* update service invo row item

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>
Signed-off-by: Shivani Singh <shivasingh@microsoft.com>

* Fixed Rego/OPA broken link

Signed-off-by: Shivani Singh <shivasingh@microsoft.com>

Signed-off-by: Nick Greenfield <nigreenf@microsoft.com>
Signed-off-by: Shivani Singh <shivasingh@microsoft.com>
Signed-off-by: Yash Nisar <yashnisar@microsoft.com>
Signed-off-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>
Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>
Co-authored-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>
Co-authored-by: Yash Nisar <yashnisar@microsoft.com>
Co-authored-by: Hannah Hunter <hannahhunter@microsoft.com>
Co-authored-by: Mark Fussell <markfussell@gmail.com>

* fix links

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

Signed-off-by: 1046102779 <seachen@tencent.com>
Signed-off-by: Adrian Cole <adrian@tetrate.io>
Signed-off-by: Crypt Keeper <64215+codefromthecrypt@users.noreply.github.com>
Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>
Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>
Signed-off-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>
Signed-off-by: yellow chicks <seachen@tencent.com>
Signed-off-by: Nick Greenfield <nigreenf@microsoft.com>
Signed-off-by: yaron2 <schneider.yaron@live.com>
Signed-off-by: Lorenzo Montanari <l0ll098@users.noreply.github.com>
Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>
Signed-off-by: Loong <loong.dai@intel.com>
Signed-off-by: RcXu <honeyxhl@163.com>
Signed-off-by: Greg. A. <gautric@users.noreply.github.com>
Signed-off-by: Yash Nisar <yashnisar@microsoft.com>
Signed-off-by: Maarten Mulders <mthmulders@users.noreply.github.com>
Signed-off-by: Sarthak Sharma <sartsharma@microsoft.com>
Signed-off-by: Mark Fussell <markfussell@gmail.com>
Signed-off-by: Shivani Singh <shivasingh@microsoft.com>
Co-authored-by: 1046102779 <seachen@tencent.com>
Co-authored-by: Adrian Cole <adrian@tetrate.io>
Co-authored-by: Crypt Keeper <64215+codefromthecrypt@users.noreply.github.com>
Co-authored-by: Mark Fussell <markfussell@gmail.com>
Co-authored-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>
Co-authored-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>
Co-authored-by: Yaron Schneider <schneider.yaron@live.com>
Co-authored-by: Nick Greenfield <nigreenf@microsoft.com>
Co-authored-by: Lorenzo Montanari <l0ll098@users.noreply.github.com>
Co-authored-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>
Co-authored-by: Loong Dai <loong.dai@intel.com>
Co-authored-by: Mr.Black <honeyxhl@163.com>
Co-authored-by: Greg. A <gautric@users.noreply.github.com>
Co-authored-by: Yash Nisar <yashnisar@microsoft.com>
Co-authored-by: Maarten Mulders <mthmulders@users.noreply.github.com>
Co-authored-by: Boris Lublinsky <blublinsky@hotmail.com>
Co-authored-by: Sarthak Sharma <sartsharma@microsoft.com>
Co-authored-by: shivanisinghnitp <58537821+shivanisinghnitp@users.noreply.github.com>

nyemade-uversky reviewed (changes requested):
@mcandeia could you move these docs out of the developing-applications directory and into the operations directory instead?

tmacam commented on daprdocs/content/en/operations/extending/_index.md:
```diff
@@ -0,0 +1,7 @@
+---
+type: docs
+title: "Extending dapr"
+linkTitle: "Extending dapr"
```
Dapr is always capitalized in docs. This applies to all other docs in this PR.

```suggestion
title: "Extending Dapr"
linkTitle: "Extending Dapr"
```

tmacam commented on daprdocs/content/en/operations/extending/developing-grpc-components.md:
```diff
@@ -0,0 +1,339 @@
+title: "Developing gRPC-based Pluggable Components"
+linkTitle: "Developing gRPC-based Pluggable Components"
+weight: 250
+description: "Extending dapr with external gRPC-based components"
```
```suggestion
description: "Extending Dapr with external gRPC-based components"
```

tmacam commented on daprdocs/content/en/operations/extending/developing-grpc-components.md:
```diff
@@ -0,0 +1,339 @@
+description: "Extending dapr with external gRPC-based components"
+---
+
+[gRPC-based](https://grpc.io/) Dapr components are typically run as containers or processes that communicate with the dapr main process via [Unix Domain Sockets](https://en.wikipedia.org/wiki/Unix_domain_socket).
```
It would be better to create context to state what gRPC Dapr components _are_ instead of saying _how_ they are typically used. As this is the first document in this series, this document should provide context of what is to follow if the reader choose to embark in this reading journey - or link to some place that would provide such.

Suggestion: Use reference links to avoid repeating the main GRPC link. Not really a big deal as you only link it twice but it would allow you to easily turn any textual reference to gRPC into a links.

mcandeia replied:
Agree, will work on that

tmacam commented on daprdocs/content/en/operations/extending/using-grpc-components.md:
```diff
@@ -0,0 +1,87 @@
+title: "Using gRPC-based Pluggable Components"
+linkTitle: "Using gRPC-based Pluggable Components"
+weight: 250
+description: "Extending dapr with external gRPC-based components"
```
```suggestion
description: "Extending Dapr with external gRPC-based components"
```

tmacam commented on daprdocs/content/en/operations/extending/using-grpc-components.md:
```diff
@@ -0,0 +1,87 @@
+description: "Extending dapr with external gRPC-based components"
+---
+
+For more information on how to develop gRPC-based Pluggable Components click [here]({{< ref developing-grpc-components.md >}}).
+
```
```suggestion
For more information on how to develop gRPC-based Pluggable Components check ["Developing gRPC-based Pluggable Components"]({{< ref developing-grpc-components.md >}}).

```

tmacam commented on daprdocs/content/en/operations/extending/using-grpc-components.md:
```diff
@@ -0,0 +1,87 @@
+
+For more information on how to develop gRPC-based Pluggable Components click [here]({{< ref developing-grpc-components.md >}}).
+
+The example above shows how a state store gRPC-based pluggable component is declared and used, the same can be applied for the others components.
```
"Above"? As this is a single document, referencing the previous document by "above" is ambiguous as there is no content or example preceding this section of the document.

I would be better to either: 
* directly link to it or 
* assume a read ordering and taking this to be the second document read, use _previous_ instead.

mcandeia replied:
this text shouldn't be there, deleted, thanks

mcandeia replied:
removed!

tmacam commented on daprdocs/content/en/operations/extending/developing-grpc-components.md:
```diff
@@ -0,0 +1,339 @@
+
+### Step 1: Prerequisites
+
+For this tutorial, you must have minimal knowledge of [gRPC and protocol buffers](https://grpc.io/), and that you chose a programming language [that supports gRPC](https://grpc.io/docs/languages/).
```
```suggestion
For this tutorial, you must:
1. Have some knowledge of how to use and build services using [gRPC and protocol buffers](https://grpc.io/). In doubt, check the their [Quick Start](https://grpc.io/docs/languages/csharp/quickstart/) and [Basics tutorial](https://grpc.io/docs/languages/csharp/quickstart/) .
2. Use a [gRPC-supported programming language](https://grpc.io/docs/languages/).
```

tmacam commented on daprdocs/content/en/operations/extending/developing-grpc-components.md:
```diff
@@ -0,0 +1,339 @@
+
+### Step 1: Prerequisites
+
+For this tutorial, you must have minimal knowledge of [gRPC and protocol buffers](https://grpc.io/), and that you chose a programming language [that supports gRPC](https://grpc.io/docs/languages/).
+
+For simplicity, all code samples uses the generated code from the [protoc](https://developers.google.com/protocol-buffers/docs/downloads) tool or supported gRPC build tools by languages.
+
+As previously mentioned, dapr uses a Unix Domain Socket to communicate with gRPC-based components, which means that as a prerequisite you also need a UNIX-like system, or for Windows users, [WSL](https://learn.microsoft.com/en-us/windows/wsl/install) should be sufficient.
```
I think we could move this particular paragraph to a language-agnostic section and provide links for .Net specific stuff later. What you think?

mcandeia replied:
Not sure, [I've found other documentation doing the same](https://github.com/dapr/docs/blob/v1.8/daprdocs/content/en/getting-started/quickstarts/serviceinvocation-quickstart.md), and I think it worth to pay for duplicated text against having two prerequisite sections?

tmacam commented on daprdocs/content/en/operations/extending/developing-grpc-components.md:
```diff
@@ -0,0 +1,339 @@
+
+As previously mentioned, dapr uses a Unix Domain Socket to communicate with gRPC-based components, which means that as a prerequisite you also need a UNIX-like system, or for Windows users, [WSL](https://learn.microsoft.com/en-us/windows/wsl/install) should be sufficient.
+
+Also, install [.NET Core 6+](https://dotnet.microsoft.com/en-us/download)
```
```suggestion
Install [.NET Core 6+](https://dotnet.microsoft.com/en-us/download)
```

tmacam commented on daprdocs/content/en/operations/extending/developing-grpc-components.md:
```diff
@@ -0,0 +1,339 @@
+
+Also, install [.NET Core 6+](https://dotnet.microsoft.com/en-us/download)
+
+This tutorial is based on the [official Microsoft documentation](https://learn.microsoft.com/en-us/aspnet/core/grpc/basics?view=aspnetcore-6.0#c-tooling-support-for-proto-files).
```
```suggestion
This tutorial is based on the [official Microsoft documentation for development using Protocol Buffers](https://learn.microsoft.com/en-us/aspnet/core/grpc/basics?view=aspnetcore-6.0#c-tooling-support-for-proto-files).
```

tmacam commented on daprdocs/content/en/operations/extending/developing-grpc-components.md:
```diff
@@ -0,0 +1,339 @@
+
+### Step 2: Downloading the base template project
+
+We've prepared a blueprint project that helps you to skip a few manual steps like cloning dapr repository and adding the required libraries. Get started by downloading and unzipping it using the following <a href="/code/dotnet-memstore.zip">link</a>.
```
```suggestion
We've prepared a blueprint project that helps you to skip a few manual steps like cloning Dapr repository and adding the required libraries. Get started by downloading and unzipping it using the following <a href="/code/dotnet-memstore.zip">link</a>.
```

tmacam commented on daprdocs/content/en/operations/extending/developing-grpc-components.md:
````diff
@@ -0,0 +1,339 @@
+At this point, you are ready to build and run the component. Let's test it out by running `dotnet build`
+
+```
+➜  DaprMemStoreComponent dotnet build
````
Nit: Not sure on windows-world but shell prompts are usually referenced by `$` instead of `➜`.

mcandeia replied:
yup, its my zsh prompt theme

tmacam commented on daprdocs/content/en/operations/extending/developing-grpc-components.md:
````diff
@@ -0,0 +1,339 @@
+Time Elapsed 00:00:01.03
+```
+
+Great!
+
+Now, let's run the StateStore service by issuing a `dotnet run`
````
```suggestion
Great! You've just built your first gRPC-based pluggable component: an in-memory StateStore component!

Now, let's run this StateStore service by issuing a `dotnet run`
```

tmacam commented on daprdocs/content/en/operations/extending/developing-grpc-components.md:
```diff
@@ -0,0 +1,339 @@
+
+We've prepared a blueprint project that helps you to skip a few manual steps like cloning dapr repository and adding the required libraries. Get started by downloading and unzipping it using the following <a href="/code/dotnet-memstore.zip">link</a>.
+
+This template contains all the required changes to start developing a gRPC-based component from scratch, including protobuf definitions and an unimplemented InMemory StateStore that you are going to implement.
```
```suggestion
This template contains all the required changes to start developing a gRPC-based component from scratch, including protobuf definitions and an unimplemented in-memory StateStore that you are going to implement.
```

tmacam commented on daprdocs/content/en/operations/extending/developing-grpc-components.md:
```diff
@@ -0,0 +1,339 @@
+
+To see it working, you can use the [grpc_cli](https://github.com/grpc/grpc/blob/master/doc/command_line_tool.md) tool for making gRPC calls. You can download and use it or use your preferred tool.
+
+Once properly downloaded and installed, open a new terminal instance, and let's invoke the `Features` method from the StateStore service, using the following command: `grpc_cli call unix:///tmp/dapr-components-sockets/memstore.sock Features ''`
```
Out of curiosity: who is setting the DAPR_... env var? I am curious how the component got the `/tmp/dapr-coomponents-socket/memstore.sock` path. It wasn't given on the previous run step.

mcandeia replied:
from template, which actually contains a ready-to-use unimplemented state store including socket handling like creating/deleting

tmacam reviewed (commented):
s/dapr/Dapr/ and some other comments.

hhunter-ms commented on daprdocs/content/en/operations/extending/using-grpc-components.md:
````diff
@@ -0,0 +1,159 @@
+  - backend
+```
+
+You should also [scope](https://docs.Dapr.io/operations/components/component-scopes/) your component to make sure that only the target configured application will try to connect with the pluggable component since it will only be running in its Deployment, otherwise the runtime will fail when initializing the component.
````
```suggestion
You should also [scope]({{< ref component-scopes >}}) your component to make sure that only the target configured application will try to connect with the pluggable component since it will only be running in its Deployment, otherwise the runtime will fail when initializing the component.
```

mcandeia replied:
thanks!

hhunter-ms commented on daprdocs/content/en/operations/extending/developing-grpc-components.md:
```diff
@@ -0,0 +1,387 @@
+
+For simplicity, all code samples uses the generated code from the [protoc](https://developers.google.com/protocol-buffers/docs/downloads) tool or supported gRPC build tools by languages.
+
+As previously mentioned, Dapr uses a Unix Domain Socket to communicate with gRPC-based components, which means that as a prerequisite you also need a UNIX-like system, or for Windows users, [WSL](https://learn.microsoft.com/en-us/windows/wsl/install) should be sufficient.
```
```suggestion
As previously mentioned, Dapr uses a Unix Domain Socket to communicate with gRPC-based components, which means that as a prerequisite you also need a UNIX-like system, or for Windows users, [WSL](https://learn.microsoft.com/windows/wsl/install) should be sufficient.
```

to fix the link validation error

hhunter-ms commented on daprdocs/content/en/operations/extending/developing-grpc-components.md:
```diff
@@ -0,0 +1,387 @@
+
+Install [.NET Core 6+](https://dotnet.microsoft.com/en-us/download)
+
+This tutorial is based on the [official Microsoft documentation for development using Protocol Buffers](https://learn.microsoft.com/en-us/aspnet/core/grpc/basics?view=aspnetcore-6.0#c-tooling-support-for-proto-files).
```
```suggestion
This tutorial is based on the [official Microsoft documentation for development using Protocol Buffers](https://learn.microsoft.com/aspnet/core/grpc/basics?view=aspnetcore-6.0#c-tooling-support-for-proto-files).
```

to fix the link validation error

mcandeia committed:
Fix tutorial links to remove lang specific

Signed-off-by: Marcos Candeia <marrcooos@gmail.com>

mcandeia committed:
Added grpc components diagram

Signed-off-by: Marcos Candeia <marrcooos@gmail.com>

yaron2 committed:
Add AWS S3 presign url (#2832)

* add aws s3 presign url

Signed-off-by: yaron2 <schneider.yaron@live.com>

* fix broken link

Signed-off-by: yaron2 <schneider.yaron@live.com>

* fix broken link 2

Signed-off-by: yaron2 <schneider.yaron@live.com>

* Update daprdocs/content/en/reference/components-reference/supported-bindings/s3.md

Co-authored-by: Mark Fussell <markfussell@gmail.com>
Signed-off-by: Yaron Schneider <schneider.yaron@live.com>

* remove filePath

Signed-off-by: yaron2 <schneider.yaron@live.com>

* add upload description

Signed-off-by: yaron2 <schneider.yaron@live.com>

* fix broken link

Signed-off-by: yaron2 <schneider.yaron@live.com>

* fix broken link

Signed-off-by: yaron2 <schneider.yaron@live.com>

* add presign existing object content

Signed-off-by: yaron2 <schneider.yaron@live.com>

Signed-off-by: yaron2 <schneider.yaron@live.com>
Signed-off-by: Yaron Schneider <schneider.yaron@live.com>
Co-authored-by: Mark Fussell <markfussell@gmail.com>

RcXu committed:
Add Alicloud SLS output binding docs (#2711)

* add alicloud sls binding component document

Signed-off-by: RcXu <honeyxhl@163.com>

* Add Alicloud SLS output binding docs

Signed-off-by: RcXu <honeyxhl@163.com>

* Add Alicloud SLS output binding docs

Signed-off-by: RcXu <honeyxhl@163.com>

* Update daprdocs/content/en/reference/components-reference/supported-bindings/alicloudsls.md

Co-authored-by: Mark Fussell <markfussell@gmail.com>
Signed-off-by: Mr.Black <honeyxhl@163.com>

* Apply suggestions from code review

Co-authored-by: Mark Fussell <markfussell@gmail.com>
Signed-off-by: Mr.Black <honeyxhl@163.com>

Signed-off-by: RcXu <honeyxhl@163.com>
Signed-off-by: Mr.Black <honeyxhl@163.com>
Co-authored-by: Mark Fussell <markfussell@gmail.com>
Co-authored-by: Hannah Hunter <94493363+hhunter-ms@users.noreply.github.com>

msfussell commented:
@mcandeia - I suggest that we talk about this PR. This is a start, however there are several things that we need to address here from samples, style, content, the positioning of this feature, presumable this is a preview feature, to location in the docs.

tmacam committed:
Update HashiCorp Vault status to Stable (#2856)

* Update HashiCorp Vault status to Beta

With dapr/components-contrib#1926, HashiCorp Vault has both conformance and certifications tests and is ready to be promoted to Beta and, on the next release, to Stable status.

Signed-off-by: Tiago Alves Macambira <tmacam@burocrata.org>

* Promoting to stable instead of beta then stable

This "stable in one release" is aligned with https://github.com/dapr/dapr/issues/4970

Signed-off-by: Tiago Alves Macambira <tmacam@burocrata.org>

Signed-off-by: Tiago Alves Macambira <tmacam@burocrata.org>
Co-authored-by: Mark Fussell <markfussell@gmail.com>

tmacam committed:
Mark Memcached as a stable component. (#2836)

On dapr/components-contrib#2116 certification tests for Memcached
state store were added to `components-contrib`, turning this component
into a stable one.

This PR updates the documentation to reflect that.

Closes dapr/components-contrib#1929

Signed-off-by: Tiago Alves Macambira <tmacam@burocrata.org>

Signed-off-by: Tiago Alves Macambira <tmacam@burocrata.org>
Co-authored-by: Hannah Hunter <94493363+hhunter-ms@users.noreply.github.com>
Co-authored-by: Mark Fussell <markfussell@gmail.com>

ItalyPaleAle committed:
Updated metadata for various Azure components (#2855)

Fixes #2613

Signed-off-by: ItalyPaleAle <43508+ItalyPaleAle@users.noreply.github.com>

Signed-off-by: ItalyPaleAle <43508+ItalyPaleAle@users.noreply.github.com>
Co-authored-by: Mark Fussell <markfussell@gmail.com>

cmendible committed:
Azure App Configuration Request Metadata (#2810)

* Azure App Configuration Request Metadata

Signed-off-by: cmendible <cmendible@gmail.com>

* Update daprdocs/content/en/reference/components-reference/supported-configuration-stores/azure-appconfig-configuration-store.md

Co-authored-by: Mark Fussell <markfussell@gmail.com>
Signed-off-by: Carlos Mendible <266546+cmendible@users.noreply.github.com>

* Update daprdocs/content/en/reference/components-reference/supported-configuration-stores/azure-appconfig-configuration-store.md

Co-authored-by: Mark Fussell <markfussell@gmail.com>
Signed-off-by: Carlos Mendible <266546+cmendible@users.noreply.github.com>

* Update daprdocs/content/en/reference/components-reference/supported-configuration-stores/azure-appconfig-configuration-store.md

Co-authored-by: Mark Fussell <markfussell@gmail.com>
Signed-off-by: Carlos Mendible <266546+cmendible@users.noreply.github.com>

* Update daprdocs/content/en/reference/components-reference/supported-configuration-stores/azure-appconfig-configuration-store.md

Co-authored-by: Mark Fussell <markfussell@gmail.com>
Signed-off-by: Carlos Mendible <266546+cmendible@users.noreply.github.com>

Signed-off-by: cmendible <cmendible@gmail.com>
Signed-off-by: Carlos Mendible <266546+cmendible@users.noreply.github.com>
Co-authored-by: Mark Fussell <markfussell@gmail.com>
Co-authored-by: Hannah Hunter <94493363+hhunter-ms@users.noreply.github.com>

ItalyPaleAle committed:
Document removal of `http-stream-request-body` (#2853)

Fix #2737

Signed-off-by: ItalyPaleAle <43508+ItalyPaleAle@users.noreply.github.com>

Signed-off-by: ItalyPaleAle <43508+ItalyPaleAle@users.noreply.github.com>
Co-authored-by: Mark Fussell <markfussell@gmail.com>

yash-nisar committed:
Include the newly added `time` field for CloudEvent (#2858)

Closes https://github.com/dapr/docs/issues/2845

Signed-off-by: Yash Nisar <yashnisar@microsoft.com>

Signed-off-by: Yash Nisar <yashnisar@microsoft.com>
Co-authored-by: Mark Fussell <markfussell@gmail.com>

tmacam committed:
Second pass and second page (#2)

* Add developing Components section

Signed-off-by: Marcos Candeia <marrcooos@gmail.com>

* Add .NET initial doc

Signed-off-by: Marcos Candeia <marrcooos@gmail.com>

* Remove pluggable component CRD docs

Signed-off-by: Marcos Candeia <marrcooos@gmail.com>

* Add zip file with quickstart code

Signed-off-by: Marcos Candeia <marrcooos@gmail.com>

* Add using grpc components on standalone mode section

Signed-off-by: Marcos Candeia <marrcooos@gmail.com>

* Add dapr runtime version note

Signed-off-by: Marcos Candeia <marrcooos@gmail.com>

* Be more impersonal

Signed-off-by: Marcos Candeia <marrcooos@gmail.com>

* SecondPas
sAndSecondPage

Signed-off-by: Tiago Alves Macambira <tmacam@burocrata.org>

Signed-off-by: Marcos Candeia <marrcooos@gmail.com>
Signed-off-by: Tiago Alves Macambira <tmacam@burocrata.org>
Co-authored-by: Marcos Candeia <marrcooos@gmail.com>

mcandeia commented:
Yes, for sure @msfussell.

I was trying to get it to a "draft" version but ready to review before scheduling a call with you,

Also I'll include @tmacam and @nyemade-uversky on that

mcandeia committed:
Removing pluggable components overview section

Signed-off-by: Marcos Candeia <marrcooos@gmail.com>

msfussell commented:
High level feedback
1) Sample code is best as a sample in the Samples repo
2) A component is a component. This is just a plugggable one. 
3) Show how to use a pluggable component first, which is the same way as today for self hosted, and only K8s annotations on Dapr make this different. The spec is the same as outline in the community call today (BTW is it component specs, not CRDs. )
4) Needs to be under components section in Operations.
-  MAYBE Overview should be separate topic.
- "How to" needs to be just steps with the right style (read the Doc guidelines) and point to the sample in the Samples repo (Includes preview setting)
5) Going forward have a Tutorial for the Component which is how the HowTo is currently written, but can do this after the v1.9 release

My suggested docs places for this update are
- Concepts https://docs.dapr.io/concepts/components-concept/ - This should include a diagram
- Overview/HowTo use a pluggable component - Under https://docs.dapr.io/operations/components/
- How to develop a pluggable component - Under https://docs.dapr.io/operations/components/
- Update the K8 annotations here https://docs.dapr.io/reference/arguments-annotations-overview/
- Update the preview section here https://docs.dapr.io/operations/support/support-preview-features/

mcandeia commented:
Awesome inputs @msfussell, thanks. I think tomorrow we can deep dive in each of those offline, 

> Sample code is best as a sample in the Samples repo

I agree with that. But following your suggestion I think the "sample" code should be complete in the sense of I don't need to guide the user throughout the implementation as the sample already covers it, which differs from the "sample" that I'm using, which is only a "template" (like dotnet templates using `dotnet new` cmd). So we don't need any code snippet for those however, is that right?

@artursouza Do you have any objection here?

> A component is a component. This is just a pluggable one.

We have been discussing it for a while and seems that Pluggable Component and Component are interchangeable since dapr initial releases like [v0.3](https://github.com/dapr/dapr/pull/872/commits/4dff9590c927b62257711d2238c99ab892cced8d#diff-5be5c18a0743a324a5c944295aec85969247817aca0476458a07f592b30c2539R5) we mentioned it several times.

Also, you can see in the dapr/dapr [README](https://github.com/dapr/dapr/blob/master/README.md#goals) that we use "Pluggable Component" term for the actual components. 

So we probably need a better way to differentiate them

> Show how to use a pluggable component first, which is the same way as today for self hosted, and only K8s annotations on Dapr make this different. 

I agree with that.

> The spec is the same as outline in the community call today (BTW is it component specs, not CRDs. )

Yes. But actually our component spec is a Kubernetes CRD. I'm using the same terminology as we use on the [component schema page](https://docs.dapr.io/operations/components/component-schema/)

For the last few things:

> Update the K8 annotations here https://docs.dapr.io/reference/arguments-annotations-overview/

There's no new annotations available. I'm using the already created ones. So I guess we don't need to update anything there, right?

> Update the preview section here https://docs.dapr.io/operations/support/support-preview-features/

I can add a new line there, but It won't require explicit opt-in as the page says, since we [decided](https://github.com/dapr/dapr/issues/5210#issuecomment-1258825526) to not have a feature flag for pluggable components given that the socket creation should be sufficient to tell dapr runtime that pluggable component feature is enabled.

might need to change the sentence "Runtime preview features require explicit opt-in in order to be used." because pluggable components doesn't follow that rule

artursouza commented:
@msfussell @mcandeia Regarding the sample code being here, it actually uncovers something else.

First, the samples repo is not for an example anymore. Remember that we changed that repo to become a full end-to-end use of Dapr and created Quickstarts for that purpose. On the other hand, this self contained example of a pluggable component does not have a place right now. The user needs to download it to make progress in the docs, otherwise, it will just be a lot of commands to recreate that.


I think this code can move to quickstarts at some point. But we don't really have a good place to put these small examples for docs.

artursouza commented:
After chatting with @mcandeia, we found a better path. We move this to dapr/samples with the following folder structure: pluggable-components/templates/dotnet/statestore


After 1.9, as we increase the number of templates, it will probably justify the creation of a templates repo in dapr-sandbox. At that point, we can also discuss if it makes sense to use Yeoman or something similar.

msfussell commented:
@mcandeia - I revised the list of topics that need to be written to have separate howto on using a component and developing a component. PTAL

> I'm using the same terminology as we use on the [component schema page](https://docs.dapr.io/operations/components/component-schema/)

That page is wrong. I tried to rid the docs of CRDs and looks like there is one left. This is to ensure that we are not saying that Dapr is bound to K8s. 

> Preview

So what prevents someone from using this in production and subsequent breaking changes? This needs to be very explicit in the docs and is currently not there.

tmacam committed:
Pluggable docs overhaul (#3)

* Upmerge - 9/21 (#2818)

* feature/helmchart: supported private helm chart repo to install dapr cluster

Signed-off-by: 1046102779 <seachen@tencent.com>

* Updates wasm middleware documentation with compilation instructions

These changes are coupled with https://github.com/dapr/components-contrib/pull/1833

Signed-off-by: Adrian Cole <adrian@tetrate.io>

* Update daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md

Co-authored-by: Mark Fussell <markfussell@gmail.com>
Signed-off-by: Crypt Keeper <64215+codefromthecrypt@users.noreply.github.com>

* feedback

Signed-off-by: Adrian Cole <adrian@tetrate.io>

* permute

Signed-off-by: Adrian Cole <adrian@tetrate.io>

* feature/helmchart: supported private helm chart repo to install dapr cluster

Signed-off-by: 1046102779 <seachen@tencent.com>

* add note to aws sns/sqs component doc; quick edit

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* change data to value

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* feature/helmchart: supported private helm chart repo to install dapr cluster

Signed-off-by: 1046102779 <seachen@tencent.com>

* Patch 3 (#2691)

* Update release version from 1.8.2 to 1.8.3

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* Update shortcode from 1.8.2 to 1.8.3

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* fix: typo

Signed-off-by: 1046102779 <seachen@tencent.com>

* feature: add a private helm repo for dapr

Signed-off-by: 1046102779 <seachen@tencent.com>

* Update install-dapr-cli.md

Signed-off-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>

* Update daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md

Co-authored-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>
Signed-off-by: yellow chicks <seachen@tencent.com>

* Fix pubsub quickstart component name to match QS source code

Signed-off-by: Nick Greenfield <nigreenf@microsoft.com>

* add output of success

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* remove word

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* add sentence

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* redis 6 update (#2718)

Signed-off-by: yaron2 <schneider.yaron@live.com>

Signed-off-by: yaron2 <schneider.yaron@live.com>

* Update Dapr version to 1.8.4 (#2714)

* Update Dapr version to 1.8.4

Signed-off-by: Nick Greenfield <nigreenf@microsoft.com>

* Update CLI version to 1.8.1

Signed-off-by: Nick Greenfield <nigreenf@microsoft.com>

Signed-off-by: Nick Greenfield <nigreenf@microsoft.com>
Co-authored-by: Yaron Schneider <schneider.yaron@live.com>

* Fixed an issue with code highlighting

Signed-off-by: Lorenzo Montanari <l0ll098@users.noreply.github.com>

* Fix links to broken middleware components

Signed-off-by: Nick Greenfield <nigreenf@microsoft.com>

* RethinkDB has no transactions, etags

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* fix a typo (#2748)

Signed-off-by: Loong <loong.dai@intel.com>

Signed-off-by: Loong <loong.dai@intel.com>

* fix typo (#2751)

Signed-off-by: RcXu <honeyxhl@163.com>

Signed-off-by: RcXu <honeyxhl@163.com>
Co-authored-by: Mark Fussell <markfussell@gmail.com>

* Update component status in v1.7

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Mark components as stable in v1.8

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Correct state azure stablestorage status in v1.7

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Update setup-sqlserver.md

Signed-off-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>

* fix some typos (#2765)

Signed-off-by: Loong <loong.dai@intel.com>

Signed-off-by: Loong <loong.dai@intel.com>

* Typo fix installtion > installation (#2726)

Signed-off-by: Greg. A. <gautric@users.noreply.github.com>

Signed-off-by: Greg. A. <gautric@users.noreply.github.com>
Co-authored-by: Mark Fussell <markfussell@gmail.com>

* Add a new version of the Dapr Diagrams deck (#2773)

Signed-off-by: Nick Greenfield <nigreenf@microsoft.com>

Signed-off-by: Nick Greenfield <nigreenf@microsoft.com>

* Fix typos in MongoDB state configuration file example (#2775)

Closes https://github.com/dapr/docs/issues/2774

Signed-off-by: Yash Nisar <yashnisar@microsoft.com>

Signed-off-by: Yash Nisar <yashnisar@microsoft.com>

* update service invo row item

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* update az cli commands

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* remove unnecessary flags

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* ignore links

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* Fix broken link

Signed-off-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>

* Fix broken link

Signed-off-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>

* remove ignore links

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* java submodule

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

* Fix two typo's (#2812)

Signed-off-by: Maarten Mulders <mthmulders@users.noreply.github.com>

Signed-off-by: Maarten Mulders <mthmulders@users.noreply.github.com>

Signed-off-by: 1046102779 <seachen@tencent.com>
Signed-off-by: Adrian Cole <adrian@tetrate.io>
Signed-off-by: Crypt Keeper <64215+codefromthecrypt@users.noreply.github.com>
Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>
Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>
Signed-off-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>
Signed-off-by: yellow chicks <seachen@tencent.com>
Signed-off-by: Nick Greenfield <nigreenf@microsoft.com>
Signed-off-by: yaron2 <schneider.yaron@live.com>
Signed-off-by: Lorenzo Montanari <l0ll098@users.noreply.github.com>
Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>
Signed-off-by: Loong <loong.dai@intel.com>
Signed-off-by: RcXu <honeyxhl@163.com>
Signed-off-by: Greg. A. <gautric@users.noreply.github.com>
Signed-off-by: Yash Nisar <yashnisar@microsoft.com>
Signed-off-by: Maarten Mulders <mthmulders@users.noreply.github.com>
Co-authored-by: 1046102779 <seachen@tencent.com>
Co-authored-by: Adrian Cole <adrian@tetrate.io>
Co-authored-by: Crypt Keeper <64215+codefromthecrypt@users.noreply.github.com>
Co-authored-by: Mark Fussell <markfussell@gmail.com>
Co-authored-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>
Co-authored-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>
Co-authored-by: Yaron Schneider <schneider.yaron@live.com>
Co-authored-by: Nick Greenfield <nigreenf@microsoft.com>
Co-authored-by: Lorenzo Montanari <l0ll098@users.noreply.github.com>
Co-authored-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>
Co-authored-by: Loong Dai <loong.dai@intel.com>
Co-authored-by: Mr.Black <honeyxhl@163.com>
Co-authored-by: Greg. A <gautric@users.noreply.github.com>
Co-authored-by: Yash Nisar <yashnisar@microsoft.com>
Co-authored-by: Maarten Mulders <mthmulders@users.noreply.github.com>

* cassanda state store cert (#2801)

Signed-off-by: Addison Juarez <adjuarez@microsoft.com>

Signed-off-by: Addison Juarez <adjuarez@microsoft.com>
Co-authored-by: Mark Fussell <markfussell@gmail.com>

* Add pluggable components to the concepts

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* Update concepts page

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* create extending dapr directory

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* Delete daprdocs/content/en/operations/Extending Dapr directory

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* create extending dapr directory

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* add pluggable component diagram

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* Delete daprdocs/content/en/operations/extending dapr directory

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* Create blank

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* Create _index.md

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* Delete blank

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* Delete pluggable component process.png

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* Add files via upload

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* Create overview.md

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* Create developing-pluggable-components.md

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* Update developing-pluggable-components.md

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* Update developing-pluggable-components.md

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* Delete _index.md

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* Create _index.md

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* Delete developing-pluggable-components.md

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* Create developing-pluggable-components.md

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* Edited for grammar

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* updated pluggable components section title

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

* Adding a programming language agnostic file about writting pluggable components

Signed-off-by: Tiago Alves Macambira <tmacam@burocrata.org>

* last changed at Oct 7, 2022 8:38 AM, pushed by Tiago Alves Macambira

Push to GH

* last changed at Oct 6, 2022 1:18 PM, pushed by Tiago Alves Macambira

Push to GH

* last changed at Oct 7, 2022 12:21 PM, pushed by Tiago Alves Macambira

Push to GH

* last changed at Oct 7, 2022 9:14 AM, pushed by Tiago Alves Macambira

Push gp GH

Signed-off-by: 1046102779 <seachen@tencent.com>
Signed-off-by: Adrian Cole <adrian@tetrate.io>
Signed-off-by: Crypt Keeper <64215+codefromthecrypt@users.noreply.github.com>
Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>
Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>
Signed-off-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>
Signed-off-by: yellow chicks <seachen@tencent.com>
Signed-off-by: Nick Greenfield <nigreenf@microsoft.com>
Signed-off-by: yaron2 <schneider.yaron@live.com>
Signed-off-by: Lorenzo Montanari <l0ll098@users.noreply.github.com>
Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>
Signed-off-by: Loong <loong.dai@intel.com>
Signed-off-by: RcXu <honeyxhl@163.com>
Signed-off-by: Greg. A. <gautric@users.noreply.github.com>
Signed-off-by: Yash Nisar <yashnisar@microsoft.com>
Signed-off-by: Maarten Mulders <mthmulders@users.noreply.github.com>
Signed-off-by: Addison Juarez <adjuarez@microsoft.com>
Signed-off-by: Tiago Alves Macambira <tmacam@burocrata.org>
Co-authored-by: Hannah Hunter <94493363+hhunter-ms@users.noreply.github.com>
Co-authored-by: 1046102779 <seachen@tencent.com>
Co-authored-by: Adrian Cole <adrian@tetrate.io>
Co-authored-by: Crypt Keeper <64215+codefromthecrypt@users.noreply.github.com>
Co-authored-by: Mark Fussell <markfussell@gmail.com>
Co-authored-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>
Co-authored-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>
Co-authored-by: Yaron Schneider <schneider.yaron@live.com>
Co-authored-by: Nick Greenfield <nigreenf@microsoft.com>
Co-authored-by: Lorenzo Montanari <l0ll098@users.noreply.github.com>
Co-authored-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>
Co-authored-by: Loong Dai <loong.dai@intel.com>
Co-authored-by: Mr.Black <honeyxhl@163.com>
Co-authored-by: Greg. A <gautric@users.noreply.github.com>
Co-authored-by: Yash Nisar <yashnisar@microsoft.com>
Co-authored-by: Maarten Mulders <mthmulders@users.noreply.github.com>
Co-authored-by: addjuarez <addiajuarez@gmail.com>
Co-authored-by: HackMD <37423+hackmd-hub[bot]@users.noreply.github.com>

mcandeia committed:
Add fix some text on concepts page

Signed-off-by: Marcos Candeia <marrcooos@gmail.com>

greenie-msft commented:
@mcandeia it looks like your recent commit pulled in a bunch of changes that are not relevant to the pluggable components PR. Can you fix this PR so the files changed only reflect the changes related to pluggable components.

mcandeia commented:
Yes @greenie-msft, as me @nyemade-uversky and @tmacam made changes here, I'm opening another one to make sure that we change the right docs.


Closed in favor of: https://github.com/dapr/docs/pull/2875/files
